### ISSUE TITLE
Trigger "Close UI" when clicking (x) in window title bar?

### ISSUE THREAD

**Issue by orschiro** [elboulangero/goodvibes#70]
Dear @elboulangero,

I would welcome a setting by which I can simply press the (x) in the window title bar to trigger the "Close UI" action. Using `Ctrl + C` for that is not very intuitive, in my opinion.

Your thoughts?

**Comment by elboulangero** [comment]
The "explicit and intuitive" way to close the UI is to open the main menu ans click on "Close UI". I guess you already know that. As for `Ctrl + C`, I didn't even remember that shortcut.

But in short, there's already two ways to close the UI, one explicit with the mouse and one shortcut with the keyboard. Honestly, I find it enough, I don't want to make it more complicated. I'm not even fan of the "Close UI" feature to be honest, it's a bit of a hack. When you close the UI, there's no icon nowhere to bring back the UI, which is confusing.

BTW, what desktop do you use ? If you want to have Goodvibes running in "system tray" mode, you can launch it with `--status-icon`. It can make sense, or not, depends on your desktop and your taste.

**Comment by orschiro** [comment]
@elboulangero thanks for elaborating. Let me attempt to clarify. I do sympathise with a lot of your thoughts. I think the tray icon is a great idea. How about the following:

1. Add an option to Preferences > Display to enable/disable the tray icon
2. If enabled, closing (x) the UI, automatically minimises to tray
3. If disabled, closing (x) the UI, quits the application

Would this be consistent, in your opinion?

**Comment by elboulangero** [comment]
I completely agree with you, I also think that's the best way to handle that and makes different desktop users happy (hopefully). But the situation is more complicated.

The issue at the current moment is that there is already a tray-icon mode (when launching Goodvibes from the console with `--status-icon`), but in this mode Goodvibes behaves quite differently. It will not show a main window, but display the window as a popup menu when you click on the tray icon. You can try, it's quite cool :) I call that the "old-school" mode.

The short story is that, when I started developing Goodvibes, I was using Openbox, and I wanted this particular behavior. Later on, I realized that it can only work using a feature of GTK+ (GtkStatusIcon) that is being removed (it's deprecated in GTK+3, and it's gone in GTK+4). Plus, GNOME is also removing status icons from the desktop. Other desktops (Ubuntu/KDE) use a replacement (appindicator) that only allow to display a menu when you click on the tray icon, not a whole window like I do currently.

So basically, there's no future for the tray icon mode that I implemented in Goodvibes. I started in the wrong direction.

Right now Goodvibes still support this "old-school" mode for the tray icon. I don't want to change this feature right now, because there might be people using it. GtkStatusIcon is not completely dead, as long as GTK+3 is there and there are desktops still supporting it (Openbox and plenty of others). And as long as this feature is there, I can't implement an alternative status icon mode like you suggest, it would be too confusing for everyone (including me, I see headaches coming).

My plan is to remove this "old-school" mode when I port Goodvibes to GTK+4. It won't happen soon (GTK+4 is still being developped). But when it happens, and after I get rid of this old-school mode, then it will be possible to implement the tray icon like you suggest.

It might take years :)

In the meantime, you will have to live with the `Ctrl + C` thing I'm afraid...

**Comment by orschiro** [comment]
Wow, thanks for your thorough elaboration! Quite interesting to get a behind the scenes look. :-)

Let's see what the future will bring. Just wanted to let you know that Gradio handles this quite elegantly. 

You play the radio, you close (x) the UI and Gradio magically keeps running the background without even showing a tray icon. Then you can just normally open it again from your start menu and the whole UI is there again. Quite neat. :-)

**Comment by elboulangero** [comment]
Hmm so GRadio do it by default... I didn't want to do that because I'm afraid some users freak out when they close the app and the music keeps playing, and don't realize that they need to open the app again and close it through the menu. Maybe I'm a bit more conservative than GRadio.

In any case, this will be reworked in a future version of Goodvibes after I get rid of the current tray icon behavior. So we'll keep this ticket open.
